providers/azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/strategy/CleanupResources.java

```diff
@@ -171,9 +216,12 @@ public boolean cleanupSecurityGroupIfOrphaned(String resourceGroup, String group
                logger.debug(">> deleting orphaned security group %s from %s...", name, resourceGroup);
                try {
                   deleted = resourceDeleted.apply(sgapi.delete(name));
+                  resourceRemoved.apply(IdReference.create(securityGroup.id()));
```

Comment:
Then, please add a comment in places where we are using this, to make explicit this is done to deal with eventual consistency issues. Otherwise, we might be tempted to accidentally change this in the future.